doc/programming/parsing.rst

```diff
@@ -1009,17 +1009,112 @@ Generally, a field ``bitfield(N)`` field is parsed like an
 ``uint<N>``. The field then supports dereferencing individual bit
 ranges through their labels. The corresponding expressions
 (``self.x.<id>``) have the same ``uint<N>`` type as the parsed value
-itself, with the value shifted to the right so that the lowest
-extracted bit becomes bit 0 of the returned value. As you can see in
+itself, with the value shifted to the right so that the least significant
+extracted bit becomes the least significant bit of the returned value. As you can see in
 the example, the type of the field itself becomes a tuple composed of
 the values of the individual bit ranges.
 
 By default, a bitfield assumes the underlying integer comes in network
 byte order. You can specify a ``&byte-order`` attribute to change that
 (e.g., ``bitfield(32) { ... } &byte-order=spicy::ByteOrder::Little``).
-Furthermore, each bit range can also specify a ``&bit-order``
-attribute to specify the :ref:`ordering <spicy_bitorder>` for its
-bits; the default is ``spicy::BitOrder::LSB0``.
+
+..
+    This doesn't actually have any effect.
+
+    Furthermore, each bit range can also specify a ``&bit-order``
+    attribute to specify the :ref:`ordering <spicy_bitorder>` for its
+    bits; the default is ``spicy::BitOrder::LSB0``.
+
+When parsing a ``bitfield(16)`` in the network byte order and ``spicy::BitOrder::LSB0``,
+the bits are numbered 0 to 15 from right to left.
+
+.. code::
+
+    MSB                           LSB
+         <--    1        <--        0
+    6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0
+    +---------------+---------------+
+    |     byte 1    |   byte 0      |
+    +-------------------------------+
+
+
+This default layout may be surprising given that some RFCs use inverse bit
```

Comment:
If it's indeed surprising, I'm wondering if we should change the default (though that would break things of course).

Reply:
Hmm.. Maybe as a module level specification?

Let me address the other points. Thanks!